source/analysis/AM1DNormalizationAB.h

```diff
@@ -130,6 +130,8 @@ protected slots:
 
 	/// Flag for knowing whether we need to compute the values.
 	mutable bool cacheUpdateRequired_;
+	/// A list of indices of values that need to be updated.  These are meant for when scans are running and data is coming in a pixel at a time.
+	mutable QList<AMnDIndex> dirtyIndices_;
```

Comment:
Ah, you have noticed the same thing @helfrij did in my last big analysis block update.  Yes, you are 100% correct that a lot of this could be done in a subclass.  However, as it stands right now, it would not be able to be in AMStandardAnalysisBlock because not all AB use this caching method (mostly old ones that are there for legacy reasons).  And unfortunately, AMCachedAnalysisBlock that would inherit from AMStandardAnalysisBlock would take more effort than I am willing at the moment because all AB's are AMDbObjects and every AB that has been modified would need a database upgrade (or one massive one that updated the database with the new schema).  I would love, more than anything, to do a code refactor of our analysis blocks.  Remove the old ones, update the current ones, and just make the whole system a little more robust.